src/Logic.hs

```diff
@@ -817,9 +835,14 @@ tryIntegratePullRequest pr state =
       Right (Sha sha) -> do
         -- If it succeeded, set the build to pending,
         -- as pushing should have triggered a build.
+        let Pr.MandatoryChecks check = Pr.mandatoryChecks state
+            integrationStatus = if Set.null check
+              then Pr.AnyCheck BuildPending
+              else Pr.SpecificChecks (Map.fromList (zip (Set.toList check) (repeat BuildPending)))
```

Comment:
There's `Map.fromSet` which should make this initialization slightly less convoluted.